tests/test_api.py

```diff
@@ -912,6 +913,82 @@ def __init__(self):
         self.assertEqual(m.GetAll('org.test.MyMockI'), {'blurb': 5})
 
 
+class TestServiceAutostart(dbusmock.DBusTestCase):
+    '''Test service starting DBusMockObject'''
+
+    @classmethod
+    def setUpClass(cls):
+        cls.xdg_data_dir = tempfile.mkdtemp(prefix='dbusmock_xdg_')
+        cls.addClassCleanup(shutil.rmtree, cls.xdg_data_dir)
+
+        os.environ['XDG_DATA_DIRS'] = cls.xdg_data_dir
+
+        os.mkdir(os.path.join(cls.xdg_data_dir, 'dbus-1'))
+        system_dir = os.path.join(cls.xdg_data_dir, 'dbus-1', 'system-services')
+        session_dir = os.path.join(cls.xdg_data_dir, 'dbus-1', 'services')
+        os.mkdir(system_dir)
+        os.mkdir(session_dir)
+
+        with open(os.path.join(system_dir, 'org.TestSystem.service'), 'w', encoding='ascii') as s:
+            s.write('[D-BUS Service]\n' +
+                    'Name=org.TestSystem\n'
+                    'Exec=/usr/bin/python3 -c "import sys; from gi.repository import GLib, Gio; '
+                    '     Gio.bus_own_name(Gio.BusType.SYSTEM, \'org.TestSystem\', 0, None, None, lambda *args: sys.exit(0)); '
+                    '     GLib.MainLoop().run()"\n'
```

Comment:
Hah, good one! It's a pleasure how compact it is in Python.

Please use a `"""` multiline string for this, which avoids the (inconsistent) `+`, `\n`, and quoted single quote.

Reply:
ah, I tried. But the script line is way too long and then the code tests complain about that.

I could add a comment to ignore the warning instead.